### DISCUSSION TITLE
How to use ReactQuery without React

### DISCUSSION THREAD

**Discussion by VanTanev** [TanStack/query#2582] [General]
It would be great if there was some guidance on how to use the raw react query API, regardless of React.

1) How to create a query (observer?) and listen for changes?
2) How to mark a particular query for refetch
3) How to use MutationCache.build()

others?

**Comment by TkDodo** [comment]
> 1. How to create a query (observer?) and listen for changes?

I guess you want to subscribe to the cache? the react-query-devtools do exactly that. The most simple example would likely be how `useIsFetching` subscribes: https://github.com/tannerlinsley/react-query/blob/4a821b0f1f94fecdc0c19d1d914aa1f4a2cc0e6d/src/react/useIsFetching.ts#L31-L49

If you really need an observer, have a look at `useBaseQuery`.

> 2. How to mark a particular query for refetch

`queryClient.invalidateQueries(key)`. It will also refetch the query if it has active subscribers. if you don't want that:

`queryClient.invalidateQueries(key, { refetchActive: false })`

> 3. How to use MutationCache.build()

No idea what that is 😅
